### PR TITLE
[RAPTOR-5822] add schema for remaining templates

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Part 2.  Add typeschema to tasks and estimator templates.

### CODE REVIEW

File: task_templates/estimators/python_anomaly/model-metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: python_anomaly
+type: training
+targetType: anomaly
+# modelID: optional
+environmentID: 5e8c889607389fe0f466c72d
+#trainingModel:
+#  trainOnProject: optional project on which to train
+validation:
+  # Path is relative to this file
+  input: ../../../tests/testdata/boston_housing.csv
```
Comment (jmbannon) [inline task_templates/estimators/python_anomaly/model-metadata.yaml:10]: doesn't this have text? We should make sure these validation ones work

File: task_templates/estimators/r_binary_classification/model-metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: r_binary_classification
+type: training
+targetType: binary
+# modelID: optional
+environmentID: 5e8c889607389fe0f466c72d
```
Comment (jmbannon) [inline task_templates/estimators/r_binary_classification/model-metadata.yaml:5]: this needs to match the recent R public env ID

File: task_templates/estimators/r_regression/model-metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: r_regression
+type: training
+targetType: regression
+# modelID: optional
+environmentID: 5e8c889607389fe0f466c72d
```
Comment (jmbannon) [inline task_templates/estimators/r_regression/model-metadata.yaml:5]: R env id

**Review by jmbannon — CHANGES_REQUESTED** [review]
Lets set the validation input to whatever the functional test uses for each template. And make sure R uses the R env ID

**Comment by scottp-ml** [comment]
@jmbannon The issue with setting them to accept the input of the functional tests is that these were written to ignore columns that aren't useful.  So if you add text to them, that means they would be expected to work if they only received text, which isn't true.

File: task_templates/transforms/custom_class_python/model-metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    - field: number_of_columns
+      condition: NOT_LESS_THAN
+      value: 1
+    - field: sparse
+      condition: EQUALS
+      value: FORBIDDEN
+    - field: contains_missing
+      condition: EQUALS
+      value: SUPPORTED
+  output_requirements:
```
Comment (MikhailYakubovskiy) [inline task_templates/transforms/custom_class_python/model-metadata.yaml:26]: Scott, we should also add that this task never outputs missing values

File: task_templates/estimators/r_binary_classification/model-metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: r_binary_classification
+type: training
+targetType: binary
+# modelID: optional
+environmentID: 610305345da4263681a176b5
```
Comment (jmbannon) [inline task_templates/estimators/r_binary_classification/model-metadata.yaml:5]: On staging the R environmentID is `5ea850ca1d41c8173c2feef6`

Comment (jmbannon) [reply task_templates/estimators/r_binary_classification/model-metadata.yaml:5]: https://staging.datarobot.com/model-registry/custom-environments/5ea850ca1d41c8173c2feef6

**Pull request merged**